client/src/app/pages/assessment-management/questionnaire/questionnaire-page.tsx

```diff
@@ -0,0 +1,221 @@
+  ) => {
+    setActiveTabKey(tabIndex as number);
+  };
+
+  const [assessmentData, setAssessmentData] = useState<YamlAssessment | null>(
+    null
+  );
+  const activeSection = assessmentData?.sections[activeTabKey];
+
+  // ------------------------!!
+  // TODO: replace this with the real data from the API
+  const fetchError = false;
+
+  useEffect(() => {
+    fetch("/questionnaire-data.yaml") // adjust this path
+      .then((response) => response.text())
+      .then((data) => {
+        const parsedData = yaml.load(data) as YamlAssessment;
+        setAssessmentData(parsedData);
+      });
+  }, []);
+  // ------------------------!!
+  const tableControls = useLocalTableControls({
+    idProperty: "formulation",
+    items: activeSection?.questions || [],
+    columnNames: {
+      formulation: "Name",
+    },
+    hasActionsColumn: true,
+    isSelectable: false,
+    expandableVariant: "single",
+  });
+  const {
+    currentPageItems,
+    numRenderedColumns,
+    propHelpers: {
+      tableProps,
+      getThProps,
+      getTdProps,
+      getExpandedContentTdProps,
+    },
+    expansionDerivedState: { isCellExpanded },
+  } = tableControls;
+
+  return (
+    <>
+      <PageSection variant={PageSectionVariants.light}>
+        <TextContent>
+          <Text component="h1">Questionnaire</Text>
+        </TextContent>
+        <Breadcrumb>
+          <BreadcrumbItem>
+            <Link to={Paths.assessment}>Assessment</Link>
+          </BreadcrumbItem>
+          <BreadcrumbItem to="#" isActive>
+            {assessmentData?.name}
+          </BreadcrumbItem>
+        </Breadcrumb>
+      </PageSection>
+      <PageSection>
+        <ConditionalRender
+          // TODO: add loading state
+          // when={isFetching && !(currentPageDataFromReactQuery || fetchError)}
+          when={fetchError}
+          then={<AppPlaceholder />}
+        >
+          <div
+            style={{
+              backgroundColor: "var(--pf-v5-global--BackgroundColor--100)",
+              display: "flex",
+            }}
+          >
+            <Tabs
+              activeKey={activeTabKey}
+              onSelect={handleTabClick}
+              isVertical
+              aria-label="Tabs in the vertical example"
+              role="region"
+              width="50px"
+              className="tabs-vertical-container"
+            >
+              {assessmentData?.sections.map((section, index) => {
+                return (
+                  <Tab
+                    eventKey={index}
+                    title={
+                      <TabTitleText aria-label="vertical" role="region">
+                        {section.name}
+                      </TabTitleText>
+                    }
+                  >
+                    <TabContent
+                      id={section.name}
+                      className="tab-content-container"
+                      style={{ flex: 1 }}
+                    >
+                      <Table
```

Comment:
using the new table, nice.

would it make sense to pull this into its own component?

Reply:
Done - much cleaner page level component now.